### DISCUSSION TITLE
Import error for local npm packages

### DISCUSSION THREAD

**Discussion by gvanhorne** [observablehq/framework#1606] [Q&A] [answered]
Hello,

I'm trying to develop an npm package for use in framework apps, but when installing them locally (using `npm link`, or `npm install /path/to/package`) I get the error "TypeError: Failed to resolve module specifier 'framework-js-component'". See here for a minimal example: https://github.com/gvanhorne/framework-js-imports.

I see from the documentation that [not all node packages are usable in the browser](https://observablehq.com/framework/imports#node-imports). I suspect this is where my misunderstanding lies. Any tips for compiling/building npm packages in such a way that I can import and use them in Framework projects? Something similar to `import foo from 'framework-js-component'` in my `index.md` file.

The dev server provides some more detail: 
```
Error [RollupError]: Could not resolve entry module "framework-js-component".
    at getRollupError (file:///home/gvanhorne/workspace/framework-js-imports/framework-app/node_modules/rollup/dist/es/shared/parseAst.js:392:41)
    at error (file:///home/gvanhorne/workspace/framework-js-imports/framework-app/node_modules/rollup/dist/es/shared/parseAst.js:388:42)
    at ModuleLoader.loadEntryModule (file:///home/gvanhorne/workspace/framework-js-imports/framework-app/node_modules/rollup/dist/es/shared/node-entry.js:19262:20)
    at async Promise.all (index 0) {
  code: 'UNRESOLVED_ENTRY'
```

Any help would be appreciated, thanks! 🙏

**Comment by Fil** [comment, marked as answer]
I think it's that we don't follow the npm (or Yarn) links, because if I copy your module into node_modules, it works.

**Reply by gvanhorne** [reply]
I appreciate you looking into this and thanks for your reply! Looking into it now seems like `npm install /path/to/module --install-links` install the package normally (i.e. _not_ as a symlink). So that works for me. Is there any interest in having support for sym links as a feature request in future, or is this working as intended?

**Reply by Fil** [reply]
I don't know… maybe if there is a dead simple solution? It doesn't feel like a pressing concern (although you tripped on it, sorry for that!), so in any case I wouldn't want it to add complexity.

**Reply by gvanhorne** [reply]
No problem! In that case maybe I'll take a quick look to see if there is a quick + easy fix and otherwise am happy to leave it as is. Thanks again 🤗

**Comment by gvanhorne** [comment]
I wanted to add for anyone else experiencing this issue in future that `npm install /path/to/package --install-links` will install the package directly without a sym link and so doesn't throw the import error as discussed.

**Comment by mshafir-veson** [comment]
Hello, we are using observable framework in a pnpm monorepo as an easy way to build internal tooling around some of our APIs. It works amazing well for this! However, because of this issue we are unable to share code with the observable framework package in the repo. We are trying to share some basic util functions, it sort of works when I switch the dependency from `"veson-utils": "workspace:*"` to `"veson-utils": "file:../../libs/veson-utils"` but having the file dependency causes some instability issues with our build. I think if symlinks were possible it would be awesome and unblock this for us. For now I think we'll have to manually copy code into the observable package.

**Comment by kyrre** [comment]
Is there any way I can force it to follow symlinks?
